pkg/kubelet/config/file_linux_test.go

```diff
@@ -374,10 +371,9 @@ func expectUpdate(t *testing.T, ch chan interface{}, testCase *testCase) {
 				}
 			}
 
-			testCase.lock.Lock()
-			defer testCase.lock.Unlock()
 			if !apiequality.Semantic.DeepEqual(testCase.expected, update) {
-				t.Fatalf("%s: Expected: %#v, Got: %#v", testCase.desc, testCase.expected, update)
+				t.Logf("%s: Awaited: %#v, Got: %#v", testCase.desc, testCase.expected, update)
+				continue
```

Comment:
what spurious event are we getting here? can we adjust the test to avoid that rather than loosening this check?

Reply:
https://github.com/kubernetes/kubernetes/blob/a517eccd9f990c3f08eb0580312e08121808b79a/pkg/kubelet/config/file.go#L92-L115

Except for the watch events, kubelet periodically generates `podUpdate` events on every file to ensure to apply any changes.
See `<-listTicker.C`

I couldn't find a better workaround to avoid this yet.

Reply:
Tests trying to exercise the filesystem events should set the period to a very long value when constructing NewSourceFile to avoid that spuriously interfering.

Reply:
Setting the period to a very long value fails the test, for the case that watches directory and uses symlink,

See, 
* `dir-test/` (watched dir)
    *  `test_pod_manifest` -> `(...)linked-test/test_pod_manifest` (symlink)
* `linked-test/`
    * `test_pod_manifest` (write to the file that the symlink points to)

There is no inotify event on `dir-test` because `inotify will return events for the directory itself, and for files inside the directory` but not for the files they point to.(not recursively)
https://linux.die.net/man/7/inotify

---

@liggitt 
I could split the test to cover the case but I think ensuring the change to arrive eventually is enough for real usage.
What do you think?

Reply:
@liggitt 
Added `TestWatchFileChangedByFsNotify` to test inotify events.